### PR TITLE
OVDB-12: Add more traits for quaternions and matrices

### PR DESCRIPTION
**Pull request by danrbailey**

### CODE REVIEW

File: openvdb/Types.h [hunk 1]
```diff
@@ -212,27 +212,93 @@ using PointDataIndex64 = PointIndex<Index64, 1>;
...
 
 
-template<typename T> struct VecTraits {
+namespace internal {
+
+/// Helper metafunction used to determine if the first template parameter is a
+/// specialization of the class template given in the second template parameter
+
+template <typename T, template <typename...> class Template>
+struct SpecializationOf : std::false_type {};
```
Comment (pcucka) [inline openvdb/Types.h]: This is a generally useful metafunction, so I'd say there's no need to flag it as internal.  But our (admittedly loosely followed) naming convention is for predicates to read as questions.  So please rename this from `openvdb::internal::SpecializationOf` to `openvdb::IsSpecializationOf`.

File: openvdb/Types.h [hunk 2]
```diff
@@ -212,27 +212,93 @@ using PointDataIndex64 = PointIndex<Index64, 1>;
 ////////////////////////////////////////
 
 
-template<typename T> struct VecTraits {
+namespace internal {
+
+/// Helper metafunction used to determine if the first template parameter is a
```
Comment (pcucka) [inline openvdb/Types.h]: Flag this comment as `@brief`

File: openvdb/doc/changes.txt [hunk 3]
```diff
@@ -6,6 +6,15 @@
 @par
 <B>Version 6.0.1</B> - <I>In development</I>
 
+@par
+New features:
+- Added new @vdblink::QuatTraits QuatTraits@endlink,
+  @vdblink::MatTraits MatTraits@endlink and
+  @vdblink::ValueTraits ValueTraits@endlink type traits to complement
+  @vdblink::VecTraits VecTraits@endlink.
+- Added an @vdblink::IsSpecializationOf @endlink helper metafunction for
```
Comment (pcucka) [inline openvdb/doc/changes.txt]: The link should read "`@vdblink::IsSpecializationOf IsSpecializationOf@endlink`".

Comment (pcucka) [inline openvdb/doc/changes.txt]: The link should read "`@vdblink::IsSpecializationOf IsSpecializationOf@endlink`".

Comment (danrbailey) [reply openvdb/doc/changes.txt]: @pcucka - Done, thanks!

I also added a CI job to automate checking the Doxygen - https://github.com/AcademySoftwareFoundation/openvdb/pull/335

**Pull request merged**